lib/xclarity_client/configuration.rb

```diff
@@ -3,17 +3,21 @@
 module XClarityClient
   class Configuration
 
-    attr_accessor :username, :password, :host, :csrf_token, :auth_type, :generated_token, :verify_ssl
+    attr_accessor :username, :password, :host, :csrf_token, :auth_type, :generated_token, :port, :verify_ssl
 
     def initialize(args)
 
       args.each { |key, value| send("#{key}=", value) }
 
-      if [username, password, host, verify_ssl].any? { |item| item.nil? }
-        $lxca_log.error "XClarityClient::Configuration initialize","username, password, host, and verify_ssl must all be specified"
-        raise ArgumentError, "username, password, host, and verify_ssl must all be specified"
+      if [username, password, host, port, verify_ssl].any? { |item| item.nil? }
+        $lxca_log.error "XClarityClient::Configuration initialize","username, password, host, port and verify_ssl must all be specified"
+        raise ArgumentError, "username, password, host, port and verify_ssl must all be specified"
       end
 
+      if ( not is_valid_port(port) ) or host == ""
+        $lxca_log.error "XClarityClient::Configuration initialize","port number  and host must be valid"
```

Comment:
I think this PR should't cover this change to #{self} because it is in the whole client, another PR should be only created with these changes, if necessary.